src/sidebar/util/dom.js

```diff
@@ -21,3 +21,24 @@ export function listen(element, events, listener, { useCapture = false } = {}) {
     );
   };
 }
+
+/**
+ * Obtain the pixel height of the element with id `elementId`, including
+ * top and bottom margins.
```

Comment:
A caveat with extracting this function is that the caller must know that margin collapsing isn't being used, otherwise they could end up double-counting margins. I don't think margins are being collapsed in the context of `ThreadList`?